### PR TITLE
Make use of the [@@immediate] information when checking [@@unboxed]

### PR DESCRIPTION
**Pull request by damiendoligez**
When checking whether the `[@@unboxed]` annotation on a type is acceptable, we can make use of the information given by `[@@immediate]` annotations on other types.

Suggested by Stephan Muenzel.

See also [MPR#7511](https://caml.inria.fr/mantis/view.php?id=7511) [GPR#606](https://github.com/ocaml/ocaml/pull/606) [GPR#1133](https://github.com/ocaml/ocaml/pull/1133).

### CODE REVIEW

**Comment by alainfrisch** [comment]
(Also related to #1262.)

**Review by alainfrisch — APPROVED** [review]
LGTM

**Review by gasche — APPROVED** [review]
Someone has to update the Changes entry before this gets merged. @alainfrisch, are you planning to merge now?

**Comment by gasche** [comment]
I just rebased the PR and will merge in trunk if the CI passes.

**Comment by trefis** [comment]
Seems like you had forgotten the Changes update before merging.
I pushed https://github.com/ocaml/ocaml/commit/1707e24f0001a122f6e71eba237b1044e5f0bfa1 to fix that.

**Pull request merged**
